### PR TITLE
throwing in postStop should be more consistent

### PR DESCRIPTION
**Pull request by ktoso**
Please check the semantics first (the test), if we agree that that's what we want I'll clean up impl a bit.
The same special casing has to be there as otherwise we'd explode when the postStop would return a `same`; not sure how to best solve that there (likely other cases there as well?)

Resolves https://github.com/akka/akka/issues/24725 eventually

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala [hunk 1]
```diff
@@ -154,15 +156,48 @@ import scala.util.control.NonFatal
   }
 
   override def postStop(): Unit = {
-    behavior match {
-      case null                   ⇒ // skip PostStop
-      case _: DeferredBehavior[_] ⇒
-      // Do not undefer a DeferredBehavior as that may cause creation side-effects, which we do not want on termination.
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: this feels scary to change for the reason mentioned in this comment, e.g. spawning a child

Comment (patriknw) [reply akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala]: should we perhaps throw immediately if `Behavior.stopped(postStop)` is used with a `DeferredBehavior` as the `postStop`?

**Review by patriknw — COMMENTED** [review]
What is the use case for Behaviors.setup as a `postStop` in the first place? Should we simply not support that?

**Comment by ktoso** [comment]
Will get back to this on upcoming monday

**Review by johanandren — APPROVED** [review]
LGTM

**Review by chbatey — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
Still has WIP in the title so i'll let you merge it @ktoso :)

**Pull request merged**